enhancements/update/cvo-log-level-api.md

```diff
@@ -0,0 +1,213 @@
+* As an OpenShift engineer, I want to increase the log level of the CVO for the CI runs, so that I can more easily troubleshoot any potential issues that occurred.
+
+### Goals
+
+Add a user-facing API for controlling the run-time verbosity of the [CVO](https://github.com/openshift/cluster-version-operator).
+
+### Non-Goals
+
+Change the default logging verbosity of the Cluster Version Operator in production OCP clusters.
+
+## Proposal
+
+This enhancement proposes to add the field `OperatorLogLevel` to the structure [`ClusterVersionSpec`](https://github.com/openshift/api/blob/c3f7566f6ef636bb7cf9549bf47112844285989e/config/v1/types_cluster_version.go#L40-L96) that is used in the structure [`ClusterVersion`](https://github.com/openshift/api/blob/c3f7566f6ef636bb7cf9549bf47112844285989e/config/v1/types_cluster_version.go#L18-L34) 
+that represents the configuration for the CVO.
+This new field is named exactly after the existing corresponding field [`OperatorLogLevel`](https://github.com/openshift/api/blob/36ce464529eb357673342c06be5886c5463cfc50/operator/v1/types.go#L64-L71) of type [`LogLevel`](https://github.com/openshift/api/blob/36ce464529eb357673342c06be5886c5463cfc50/operator/v1/types.go#L91-L107) that is used inside the structure [`OperatorSpec`](https://github.com/openshift/api/blob/36ce464529eb357673342c06be5886c5463cfc50/operator/v1/types.go#L49-L89)
+by Cluster Operators. 
+The type of the newly created `OperatorLogLevel` field of the CVO **cannot** be of the same type as above mentioned [`LogLevel`](https://github.com/openshift/api/blob/36ce464529eb357673342c06be5886c5463cfc50/operator/v1/types.go#L91-L107), which is defined in the `github.com/openshift/api/operator/v1` package.
+Importing the `github.com/openshift/api/operator/v1` package from inside the `github.com/openshift/api/config/v1` package causes import cycles.
+To ensure consistency across the operators and to not cause import cycles, this type will be redefined in the 
+[api/config/v1/types_cluster_version.go](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go) file.
+
+This enhancement is thus proposing to modify the [api/config/v1/types_cluster_version.go](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go) file as described by the following text.
+
+A new field `OperatorLogLevel` will be added to the [`ClusterVersionSpec`](https://github.com/openshift/api/blob/c3f7566f6ef636bb7cf9549bf47112844285989e/config/v1/types_cluster_version.go#L40-L96) structure:
+
+```go
+type ClusterVersionSpec struct {
+	...
+	// operatorLogLevel controls the logging level of the cluster version operator.
+	//
+	// Valid values are: "Normal", "Debug", "Trace", "TraceAll".
+	// Defaults to "Normal".
+	// +optional
+	// +kubebuilder:default=Normal
+	OperatorLogLevel CVOLogLevel `json:"operatorLogLevel,omitempty"`
```

Comment:
I'm sorry if I come late to the party but I wonder whether we should future-proof a bit and carve out a substruct that clearly configures CVO (as opposed to "configures cluster"), to have a nested place for future CVO knobs?

Something like:

```
type ClusterVersionSpec struct {
	...
        OperatorOptions CVOOptions 
}
type CVOOptions struct {
	// operatorLogLevel controls the logging level of the cluster version operator.
	OperatorLogLevel CVOLogLevel `json:"operatorLogLevel,omitempty"`
}
```

Reply:
A great idea, thank you!